app/src/main/kotlin/cz/covid19cz/erouska/ui/dashboard/DashboardFragment.kt

```diff
@@ -335,6 +351,24 @@ class DashboardFragment : BaseFragment<FragmentDashboardPlusBinding, DashboardVM
         dash_card_inactive.showOrHide(!enEnabled && (viewModel.isLocationlessScanSupported() || lsEnabled) && btEnabled)
     }
 
+    private fun checkAndShowOrHideHowItWorksNotification(show: Boolean) {
+        how_it_works_container.showOrHide(
+            show &&
+                    !data_notification_container.isVisible &&
+                    !exposure_notification_container.isVisible
+        )
+    }
+
+    private fun showOrHideDataNotification(show: Boolean) {
+        data_notification_container.showOrHide(show)
+        viewModel.checkAndShowOrHideHowItWorksNotification() // check if How It Works in-app notification should be shown
```

Comment:
Tohle je zvláštní - ViewModel volá event a Fragment na základě něho zpátky volá ViewModel. Logika by měla být na jednom místě - tzn. ViewModel si pořeší co potřebuje a pak pošle event View aby se něco změnilo.

Reply:
Jo je OK to volat i z View na základě akce uživatele, ale přijde mi divné to volání z ViewModelu přes View zpátky do ViewModelu. Prostě ve MVVM architektuře má View volat metody ViewModelu podle akcí uživatele a zpátky se updatovat podle observable stavu. Ale dělat tam takovýhle loop mi přijde nebezpečný. Nešlo by to nějak rozdělit na dvě metody? Ale není blocker pro merge, spíš NIT.